syntax = "proto3";

package snowflake;

service SnowFlake {
    rpc get_unique_url_id(Request) returns (UniqueUrlID);
    rpc get_unique_user_id(Request) returns (UniqueUserID);
}

message Request {

}

// UniqueUrlID is for representing unique shortened url ids.
message UniqueUrlID {
    string value = 1;
}

// UniqueUserID is for representing unique user ids. This will be generated by snowflake service instead of
// auto increment feature of dbs.
message UniqueUserID {
    int64 value = 1;
}

